src/stores/ProductStore.ts

```diff
@@ -0,0 +1,27 @@
+import { makeAutoObservable } from "mobx";
+
+interface Product
+{
+    id: number;
+    name: string;
+    price: number;
+}
+
+class ProductStore
+{
+    products: Product[] = [
+        {id: 1 , name: 'Product 1', price: 50},
+        {id: 2 , name: 'Product 2', price: 30},
+        {id: 3 , name: 'Product 3', price: 130},
+        {id: 4 , name: 'Product 4', price: 340},
+        {id: 5 , name: 'Product 5', price: 36},
+        {id: 6 , name: 'Product 6', price: 90}
+    ];
+
+    setProducts(products: Product[]): void
```

Comment:
setProducts = (products: Product[]) => {
runInAction(()=>{
this.products.set(products)})}

Reply:
done but if you can see the logic to set the products should be done manually because mobx observable arrays do not have a set method for replacing the entire array.

runInAction: we use runInAction to ensure  that mobx observables are updated within a consistent and predictable environment, particularly when multiple observable updates need to occur together as a single atomic operation

Reply:
done but if you can see the logic to set the products should be done manually because mobx observable arrays do not have a set method for replacing the entire array.??

Reply:
@fouadkachlan